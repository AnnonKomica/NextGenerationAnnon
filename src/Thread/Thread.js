// @flow
import React from "react";
import Post from "../Post";

type Props = {
    id: string,
    posts: PostObject[],
};

type PostObject = {
    id: string,
    imgSrc?: string,
    nickName?: string,
    title?: string,
    userId: string,
    content: string,
    timestamp: string,
}

export default class Thread extends React.Component {

    props: Props;

    _renderPosts() {
        const {
            posts,
            id: threadId,
        } = this.props;

        return posts.map((post) => (
            <Post
                key={post.id}
                threadId={threadId}
                {...post}
            />
        ));
    }

    render() {
        const {id: threadId} = this.props;

        return (
            <div id={threadId} className='thread'>
                {this._renderPosts()}
            </div>
        );
    }
}
