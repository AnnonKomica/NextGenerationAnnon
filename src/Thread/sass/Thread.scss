.thread {
    height: 200px;
}
