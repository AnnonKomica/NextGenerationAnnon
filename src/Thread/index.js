// @flow
import Thread from "./Thread";

require("./sass/Thread.scss");

export default Thread;
