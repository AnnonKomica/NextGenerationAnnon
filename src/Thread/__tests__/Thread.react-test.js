import React from "react";
import renderer from "react-test-renderer";

import Thread from "../Thread";

import THREAD from "./testData";

// eslint-disable-next-line
test("Thread DOM should render correctly", () => {
    const component = renderer.create(
        <Thread id={THREAD.id} posts={THREAD.posts} />
    );

    let tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();
});
