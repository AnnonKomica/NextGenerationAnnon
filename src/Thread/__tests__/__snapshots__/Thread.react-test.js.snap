exports[`test Thread DOM should render correctly 1`] = `
<div
  className="thread"
  id="thread1">
  <div
    className="post"
    id="thread1_post0">
    <div>
      Title: 無題 User: JKwelkf23jv At: 19911209
    </div>
    <p>
      content1
    </p>
  </div>
  <div
    className="post"
    id="thread1_post1">
    <div>
      Title: 無題 User: 12kjgvsk3mc At: 19911210
    </div>
    <p>
      content2
    </p>
  </div>
</div>
`;
